src/main/java/com/venafi/vcert/sdk/connectors/tpp/ZoneConfiguration.java

```diff
@@ -45,23 +45,42 @@ public void updateCertificateRequest(CertificateRequest request) {
         subject.country(Entity.of(subject.country(), country).resolve());
         subject.province(Entity.of(subject.province(), province).resolve());
         subject.locality(Entity.of(subject.locality(), locality).resolve());
-        if(hashAlgorithm != SignatureAlgorithm.UnknownSignatureAlgorithm) {
-            request.signatureAlgorithm(hashAlgorithm);
-        } else {
-            request.signatureAlgorithm(SignatureAlgorithm.SHA256WithRSA);
+
+       // apply defaults for settings that weren't specified and then make sure they comply with policy
+       if (request.keyType() == null) {
+            request.keyType(KeyType.defaultKeyType());
+        }
+
+        switch (request.keyType())
+        {
+            case ECDSA:
+                if (request.keyCurve() == null) {
+                    request.keyCurve(EllipticCurve.ellipticCurveDefault());
+                }
+                if (request.signatureAlgorithm() == SignatureAlgorithm.UnknownSignatureAlgorithm) {
+                    request.signatureAlgorithm(SignatureAlgorithm.ECDSAWithSHA256);
```

Comment:
What about having defaultECAlg and defaultRSAAlg methods in SignatureAlgorithm class. This will allow us to easily change defaults if ever needed.